### PR TITLE
 [NET-5336] Support for templated policies

### PR DESCRIPTION
**Pull request by roncodingenthusiast**
- Add support for templated policies in role/tokens/binding-rules
- Add support for templated policies in binding rules

core prs: 
- https://github.com/hashicorp/consul/pull/18719 
- https://github.com/hashicorp/consul/pull/18708

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module github.com/hashicorp/terraform-provider-consul
 
 require (
-	github.com/hashicorp/consul/api v1.22.0
+	github.com/hashicorp/consul/api v1.10.1-0.20230928221412-f771bd38cdee
```
Comment (roncodingenthusiast) [inline go.mod:4]: temp change, will be updated once RC1 is released

**Comment by remilapeyre** [comment]
Hello @roncodingenthusiast, is the change to `go.mod` the last thing that keeps this PR in draft? If so I could make a review and wait until the RC to change the tag in `go.mod` and merge it.

**Comment by roncodingenthusiast** [comment]
> Hello @roncodingenthusiast, is the change to `go.mod` the last thing that keeps this PR in draft? If so I could make a review and wait until the RC to change the tag in `go.mod` and merge it.

yup this should be good to go. How do i go about updating docs?

**Review by remilapeyre — APPROVED** [review]
Thanks for this contribution @roncodingenthusiast !

**Pull request merged**